src/Features/Core/Portable/CodeCleanup/DiagnosticSet.cs

```diff
@@ -0,0 +1,18 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System.Collections.Immutable;
+
+namespace Microsoft.CodeAnalysis.CodeCleanup
+{
+    internal class DiagnosticSet
+    {
+        public string Description { get; }
+        public ImmutableArray<string> DiagnosticIds { get; }
+
+        public DiagnosticSet (string description, string[] diagnosticIds)
```

Comment:
1. making htis a `param array` would likely simplify a lot of the callers.
2. errant space after DiagnosticSet.

Reply:
I'll file a follow-up issue and link it to #30605 as the target PR to clean this up.